compiler/ETA/Main/DriverPipeline.hs

```diff
@@ -1305,6 +1309,7 @@ linkGeneric dflags oFiles depPackages = do
             -- TODO: Verify that the right version eta was used
             --       in the Manifests of the jars being compiled
           else return []
+    mapM_ warnHscFile $ filter (".hsc" `isSuffixOf`) (jarInputs dflags)
```

Comment:
If I pass in a `.hs` file and a `.hsc` file, the `.hsc` file ends up in `jarInputs`. If I only pass in a `.hsc` file, it will be in `oFiles`.